src/Kestrel.Core/Internal/Infrastructure/HttpUtilities.cs

```diff
@@ -29,7 +29,7 @@ public static partial class HttpUtilities
         private const ulong _http10VersionLong = 3471766442030158920; // GetAsciiStringAsLong("HTTP/1.0"); const results in better codegen
         private const ulong _http11VersionLong = 3543824036068086856; // GetAsciiStringAsLong("HTTP/1.1"); const results in better codegen
 
-        private static readonly Encoding HeaderValueEncoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: true, throwOnInvalidBytes: true);
+        private static readonly Encoding HeaderValueEncoding = new UTF8EncodingSealed();
```

Comment:
Need to store in sealed type too `private static readonly Encoding HeaderValueEncoding` else the Jit doesn't see what type it is when it uses it

Reply:
Need to store in sealed type too 
```csharp
private static readonly UTF8EncodingSealed HeaderValueEncoding
```` 
else the Jit doesn't "see" what type it is when it uses it